### ISSUE TITLE
Colored output out of order WinApi

### ISSUE THREAD

**Issue by TimonPost** [TimonPost/crossterm#38]
There is a problem occurring where WinaApi is the default like Windows 7,8.

Here's the code:

```
    let crossterm = Crossterm::new();
    ...

    println!("Currently, only Keycloak is supported as Identity Provider. When setting the");
    println!(
        "IDP URL, please note that you will have to pass {} of Keycloak.\n",
        crossterm
            .paint("the exact path to the saml client")
            .with(Color::Yellow)
    );
```

But it gets displayed as this:
![image](https://user-images.githubusercontent.com/1899599/45831396-e2d4aa00-bcc4-11e8-9440-010c64426444.png)

As you can see, the "the exact path to the saml client" gets put to the start of the message. It also seems like the color is cleared to "white" instead of the default gray.

To print an colored text in the middle of a phrase you can call `into_displayable()` on a `StyledObject`. But with WinApi the colored text will not be at the place you have placed the formatting `{}` symbols. But instead, it will be printed at the start of the console. 

_Here you have some code to reproduce._ 
```
fn main()
{
    let screen= Screen::default();
    let crossterm = Crossterm::new(&screen);

    println!(
        "Some text line 1 \n \
          Some test line to {} and more text.\n",
        crossterm
            .style("Some yellow collored text")
            .with(Color::Yellow)
            .into_displayable(&screen)
    );
}
```

I don't really know what this is I think it has to be doing something with buffering of text. Like when you call `into_dispayable()` your text will be written with WinApi. Maybe because of that WinApi will write first before rust writes the text to the console.

**Comment by TimonPost** [comment]
In version 0.5 this will be partially solved. As long as you don't use alternate screen you won't have problems with it.

**Comment by TimonPost** [comment]
This is partially solved with #44. When don't working with alternate screen it is working just fine.

**Comment by TimonPost** [comment]
This will be fully fixed in 0.9

**Comment by TimonPost** [comment]
Fixed in #115
